src/java/com/puppetlabs/ssl_utils/SSLUtils.java

```diff
@@ -298,6 +298,39 @@ public static boolean isRevoked(X509CRL crl, X509Certificate certificate) {
         return crl.isRevoked(certificate);
     }
 
+    private static X509v2CRLBuilder crlBuilder(X509CRL crl)
+            throws CRLException {
+        // The CRL is not valid if the time of checking == the time of last_update.
+        // So to have it valid right now we need to say that it was updated one second ago.
+        DateTime now = DateTime.now();
+        Date thisUpdate = now.minusSeconds(1).toDate();
+        Date nextUpdate = now.plusYears(5).toDate();
+        X509v2CRLBuilder builder =
+                new JcaX509v2CRLBuilder(crl.getIssuerX500Principal(), thisUpdate);
+        builder.setNextUpdate(nextUpdate);
```

Comment:
What is the purpose of this? I looked at https://www.bouncycastle.org/docs/pkixdocs1.4/org/bouncycastle/cert/X509v2CRLBuilder.html#setNextUpdate(java.util.Date) but still don't really understand what this means..
Also, how was 5 years chosen?

Reply:
When something pulls down this CRL (like the agent) and uses it to check that certs it's interacting with haven't been revoked, it can use this copy of the CRL until the "next update" date. After that this CRL is considered invalid. It's sort of like an expiration date, and therefore is also the date we report through the new `ca-expirations` API for the CRL.

Reply:
Interesting. Thanks!